zone-service/src/main/java/org/cloudfoundry/identity/uaa/zone/OrchestratorZoneService.java

```diff
@@ -88,13 +171,305 @@ private ConnectionDetails buildConnectionDetails(String zoneName, IdentityZone i
     private String getZoneUri(String subdomain, String uaaUri) {
         URI uaaUriObject = URI.create(uaaUri);
         String currentUAAHostName = uaaUriObject.getHost();
+        String replacement = subdomain + "." + currentUAAHostName;
+        if (StringUtils.isEmpty(subdomain)) {
```

Comment:
isEmpty is deprecated use !hasLength method

Reply:
Done. Thanks!